test/e2e/scheduling/predicates.go

```diff
@@ -1074,11 +1080,11 @@ func GetPodsScheduled(workerNodes sets.String, pods *v1.PodList) (scheduledPods,
 	for _, pod := range pods.Items {
 		if pod.Spec.NodeName != "" && workerNodes.Has(pod.Spec.NodeName) {
 			_, scheduledCondition := podutil.GetPodCondition(&pod.Status, v1.PodScheduled)
-			framework.ExpectEqual(scheduledCondition != nil, true)
-			if scheduledCondition != nil {
-				framework.ExpectEqual(scheduledCondition.Status, v1.ConditionTrue)
-				scheduledPods = append(scheduledPods, pod)
+			if scheduledCondition == nil {
+				framework.Failf("Did not find 'scheduled' condition for pod %+v", pod)
 			}
+			framework.ExpectEqual(scheduledCondition.Status, v1.ConditionTrue)
```

Comment:
`ExpectEqual` accepts an explanation message. Add one

Reply:
Convert it to the `if <condition> then framework.Failf()` style?

Reply:
Done!